shared/api/speech_features.hpp

```diff
@@ -623,7 +623,13 @@ class Phi4AudioEmbed {
 
     SpeechLibLogMel logmel;
     // already checked in Init
-    logmel.Init(sr_val == 8000? logmel_8k_attrs_: logmel_attrs_);
+    
+    // Currently we only support 8k and 16k Hz sampling rate.
+    if (sr_val != 8000 && sr_val != 16000){
+      ORTX_CXX_API_THROW("ort-extensions internal error: currently only 8k and 16k Hz sampling rate is supported. Please resample your audio file to either 8k or 16k.", ORT_RUNTIME_EXCEPTION);
```

Comment:
to be more Mobile friendly and be consistent, please try to return the error status, instead of throwing an exception.

Reply:
Also, could we avoid adding library's name in the error message reported? To the end user, it may be confusing to see ort-extensions error if there are multiple libraries involved: win-ai-gallery -> aitk -> ort-genai -> ort-extensions ...

Reply:
> Also, could we avoid adding library's name in the error message reported? To the end user, it may be confusing to see ort-extensions error if there are multiple libraries involved: win-ai-gallery -> aitk -> ort-genai -> ort-extensions ...

on calling ort-extensions from ort-genai, only error code and error message are expected. so, ort-genai can add whatever prefix for  the error message from ort-extensions.